common/BitField.h

```diff
@@ -130,10 +130,22 @@ struct BitField
   // Force default constructor to be created
   // so that we can use this within unions
   constexpr BitField() = default;
+  constexpr BitField(const BitField&) = default;
+  constexpr BitField(BitField&&) = default;
 
-  // Warning!  This copies ALL storage bits, not just the bits represented by this bitfield.
-  // This exists so that unions can be copyable.
-  BitField& operator=(const BitField&) = default;
+  // We explicitly delete the copy assignment operator here, because the
+  // default copy assignment would copy the full storage value, rather than
+  // just the bits relevant to this particular bit field.
+  // Ideally, we would just implement the copy assignment to copy only the
+  // relevant bits, but we're prevented from doing that because the savestate
+  // code expects that this class is trivially copyable.
+  BitField& operator=(const BitField&) = delete;
+  // The move-assignment operator is still supplied so that unions can be
+  // assigned to.  This still copies the full storage value, but that's the
+  // desired behavior when assigning to unions, so there's no issue with that.
+  // Note that this is still functions as a copy-assignment operator; the other
```

Comment:
> the other bitfield is not modified

Isn't it *very much* modified?

Reply:
Ah, I misread this comment. I suggest saying "right-hand operand" instead, as there's potential confusion with the other BitFields grouped in the current union.